### ISSUE TITLE
How to convert the BVLC standard caffe model file to fb?

### ISSUE THREAD

**Issue by chagyun0213** [icubecorp/nvdla_compiler#3]
Hi
Your works are great!
I built FPGA running NVDLA nv_full configuration, and confirmed ResNet-50/YOLO-tiny work well on my FPGA board using NVDLA_compiler from NVDIA. 
I like to use you compiler, but I don't know how/where to start.
If I have working ResNet50.coffemodel and ResNet50.prototxt files, how could I create FB using your tool?
Could you guide me?

-Thanks!

**Comment by icubecorp** [comment]
@chagyun0213
Hi 
our compiler now just can compiler Lenet  and run VP, because we have not FPGA can load nv_full configure. So could you try our compiler ? and it will generate a Lenet flatbuffer  defaultly, could you run the flatbuffer on your full FPGA, if it can work successfully, i think the VP full configure and FPGA full congfigure is the same env, then we can add ResNet-50 support.
thx

**Comment by chagyun0213** [comment]
Hi 
I am sorry for late response. I tested your flatbuffer(LeNet) file on FPGA. It goes through without error.
But for through test, I like to run without Softmax layer because EMU has lots of issues.
How to remove or modify to remove out the last Softmax layer?
Thanks,

**Comment by DrVijayK** [comment]
@chagyun0213
Hi
Glad to hear that you are able to run Yolo-tiny on your FPGA platform with nvdla_full configuration, with nvdla_compiler. Were you able to run on VP as well? 
We are also NVDLA users and trying to run Yolov1-tiny object detection for long time now and so far not successful. We don't have FPGA platform yet, hence trying to run on VP with nv_full configuration. The output we are getting from VP has wrong classifications, misplaced bounding boxes and very low confidence levels etc. Definitely we are missing something.  It will be of great help if you can guide us with some pointers on caffemodel, prototxt, image dimensions, any mean values subtraction, from where drivers are used from, how output dimg is interpreted etc.
Thanks a ton.

**Comment by chagyun0213** [comment]
@DrVijayK 
Hi
Based on my experience, VP and my FPGA are equal in sense of outputs.
Are you using DarkNet's pre-trained weights?

**Comment by DrVijayK** [comment]
@chagyun0213 
Hi
Thanks a lot for offering help. Yes, we are using Darknet's pre-trained cfg and weights files and converting them to prototxt and caffemodels.
And to be precise, we are using pre-trained weights file at http://pjreddie.com/media/files/yolov1/tiny-yolov1.weights

**Comment by chagyun0213** [comment]
@DrVijayK 
The problem is Leaky ReLU Layers. Replace them as ReLU and retrain. You might have to modify your converting tool also to support ReLU. Somehow Leaky ReLU doesn’t not work.

**Comment by DrVijayK** [comment]
@chagyun0213 
Thank you very much for your inputs. Highly appreciate.

**Comment by icubecorp** [comment]
> Hi
> I am sorry for late response. I tested your flatbuffer(LeNet) file on FPGA. It goes through without error.
> But for through test, I like to run without Softmax layer because EMU has lots of issues.
> How to remove or modify to remove out the last Softmax layer?
> Thanks,

@chagyun0213 ：
Hi, 
i can give a simple overview, and you can find the everyting in source code.

Lenet caffe mode layers: (------>) convert to -------> nvdla mode layers:

1.input -------> 1. nv_input
2.conv -------> 2. nv_conv + nv_sdp
3.pooling -------> 3. nv_pdp
4.conv -------> 4. nv_conv + nv_sdp
5.pooling -------> 5. nv_pdp
6.innerproduct -------> 6. nv_conv + nv_sdp
7.Relu -------> 7. nv_sdp
8.innerproduct -------> 8. nv_conv + nv_sdp
9.softmax -------> 9. nv_softmax (emu task)
the software just need fill the following struct_lists which store the nvdla_mode layers info:

1.task_list : there is 2 kinds of task, one is nvdla task which run on nvdla and the other is emu task which run on cpu. you can divide the nvdla mode layers into different kind of tasks. as this example: task0 is nvdla task which includes 2-8 and task1 is emu task which include 9.

2.memory_list: store all memory info (mem_size mem_type etc...) the layers and task need, every layers need at most 3 kind of mems: input mem, weight mem, output mem. each task need a most 6 kinds of mem: net_desc op_list op_buffer_list etc...

3.address_list: store the mem_list index

4.symbol list: store weight data from caffemode and each task info

5.tensor list: there is 2 tensors, input tensor and output tensor

6.submit list: divide the task_list into different submit

7.reloc list and event list: we don't use it yet.

and softmax just  use UMD to do a softmax, it should not have any error if the NVDLA pass. the softmax use the  innerproduct's (the layer 8) output, you can check the result of  the innerproduct  in the UMD code, it should be a array including 10 nums.

Thx

**Comment by chagyun0213** [comment]
@icubecorp 
Thanks for your explanation. That’s enought for me to understand your flow right now.
I have lots of questions, but let me try to read your codes first.
I believe LeNet is too small network to decide on. When do you release your next version that would support bigger networks like ResNet or Vgg?

Thanks!

**Comment by DrVijayK** [comment]
@chagyun0213 
No luck so far even after replacing leakyRelu with Relu in cfg and retrain. After retraining, yolo detection went through successfully, but still VP is showing incorrect object detection with very low probabilities. I am sure missing something. Any further help from you will be highly appreciated. If possible it will be of great help if you can provide a sample.

**Comment by chagyun0213** [comment]
@DrVijayK 
I recommend you dump feature maps and compare with those of Caffe. 
It will be helpful if you do layer-by-layer starting from with only input and 1st hidden layer with ReLU.

**Comment by honorpeter** [comment]
./nvdla_compiler --prototxt test_tod5.prototxt  --caffemodel VGG_TOD_SSD_300x300_iter_120000.caffemodel
(DLA) Error 0x00030003: (propagating from main.cpp, function testSetup(), line 68)
(DLA) Error 0x00030003: (propagating from main.cpp, function launchTest(), line 85)

why?
